src/magic_folder/test/cli/test_magic_folder.py

```diff
@@ -188,32 +222,47 @@ def test_joined_magic_folder(args):
         return d.addActionFinish()
 
     def get_caps_from_files(self, client_num):
-        folders = load_magic_folders(self.get_clientdir(i=client_num))
-        mf = folders["default"]
-        return mf['collective_dircap'], mf['upload_dircap']
-
-    @log_call
-    def check_config(self, client_num, local_dir):
-        mf_yaml = fileutil.read(os.path.join(self.get_clientdir(i=client_num), "private", "magic_folders.yaml"))
-        local_dir_utf8 = local_dir.encode('utf-8')
-        self.assertIn(local_dir_utf8, mf_yaml)
+        config = load_global_configuration(
+            FilePath(self.get_clientdir(i=client_num)).child("config")
+        )
+        folder_config = config.get_magic_folder("default")
+        return folder_config.collective_dircap, folder_config.upload_dircap
 
     def create_invite_join_magic_folder(self, nickname, local_dir):
-        nickname_arg = unicode_to_argv(nickname)
         local_dir_arg = unicode_to_argv(local_dir)
+        client_path = FilePath(self.get_clientdir())
+        config = client_path.child("config")
+
         # the --debug means we get real exceptions on failures
-        d = self.do_cli("magic-folder", "--debug", "create", "magic:", nickname_arg, local_dir_arg)
-        def _done(args):
+        d = self.do_cli(
+            "magic-folder", "--debug", "init",
+            "--config", config.asBytesMode().path,
+            "--listen-endpoint", "tcp:4319",
+            "--node-directory", client_path.asBytesMode().path,
+        )
```

Comment:
It'd be great if we could trash all of our application-logic tests that work via the CLI (and just leave CLI-logic tests that work via the CLI).